tests/common/runner.rs

```diff
@@ -50,30 +53,100 @@ pub fn test_execute_revm<S: Storage + Clone + Send + Sync>(
 ) {
     let concurrency_level = thread::available_parallelism().unwrap_or(NonZeroUsize::MIN);
     assert_execution_result(
-        pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
-        pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
+        &pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
+        &pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
         false, // TODO: Parameterize this
     );
 }
 
+fn calculate_receipt_root(receipt_envelopes: Vec<ReceiptEnvelope>) -> B256 {
+    // create a BTreeMap (order must be ascending)
+    let entries = receipt_envelopes
+        .into_iter()
+        .enumerate()
+        .map(|(index, receipt)| {
+            let key_buffer = alloy_rlp::encode_fixed_size(&index);
+            let mut value_buffer = Vec::new();
+            receipt.encode_2718(&mut value_buffer);
+            (key_buffer, value_buffer)
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    // calculate the MPT root
+    let mut hash_builder = alloy_trie::HashBuilder::default();
+    for (k, v) in entries {
+        hash_builder.add_leaf(alloy_trie::Nibbles::unpack(&k), &v);
+    }
+    hash_builder.root()
+}
+
+fn get_receipt_envelopes(
+    block: &Block,
+    tx_results: &[PevmTxExecutionResult],
+) -> Vec<ReceiptEnvelope> {
+    // get the receipts
+    let receipt_iter = tx_results.iter().map(|tx| tx.receipt.clone());
+
+    // get the tx_type list
+    let tx_type_iter = block
+        .transactions
+        .txns()
+        .unwrap()
+        .map(|tx| TxType::try_from(tx.transaction_type.unwrap_or_default()).unwrap());
+
+    // zip them to get the receipt envelopes
+    let receipt_envelope_iter =
+        Iterator::zip(receipt_iter, tx_type_iter).map(|(receipt, tx_type)| match tx_type {
+            TxType::Legacy => ReceiptEnvelope::Legacy(receipt.with_bloom()),
```

Comment:
Looking at this pattern we can:
- Put `tx_type_iter` first to align with the `match`.
- Move `with_bloom()`  to `receipt_iter` to make it less redundant here?